fragments/filter/README.md

```diff
@@ -0,0 +1,632 @@
+the implementation of the operators IN, BETWEEN, LIKE, and IS NULL predicates.
+
+## Grammar and schemas
+
+- The [OAFeat (CQL) spec](https://portal.ogc.org/files/96288) includes an ABNF for cql-text and both JSON Schema and OpenAPI specifications for cql-json. The standalone files are:
+  - [cql.bnf](https://github.com/opengeospatial/ogcapi-features/blob/master/extensions/cql/standard/schema/cql.bnf)
+  - [cql.json](https://github.com/opengeospatial/ogcapi-features/blob/master/extensions/cql/standard/schema/cql.json)
+  - [cql.yml](https://github.com/opengeospatial/ogcapi-features/blob/master/extensions/cql/standard/schema/cql.yml)
+- A JSON Schema for only the parts of the CQL JSON encoding required by this extension is [here](cql.json)
+- A OpenAPI specification for only the parts of the CQL JSON encoding required by this extension is [here](cql.yml)
+- xtraplatform-spatial has a CQL [ANTLR 4 grammer](https://github.com/interactive-instruments/xtraplatform-spatial/tree/master/xtraplatform-cql/src/main/antlr/de/ii/xtraplatform/cql/infra)
+
+## Queryables
+
+The Queryables mechanism allows a client to discover what variable terms are available for use when writing filter
+expressions.  These variables can be defined per-collection, and the intersection of these variables over all collections is what is available for filtering when there are no collection restrictions. These queryables are the only variables that may be used in filter expressions, and if any variable is used in expression that is not defined as a queryable, a 400 Bad Request exception should be returned.
+
+Implementers should add queryables for all root Item fields (e.g., id, collection, geometry) with those names. Fields in Item Properties should also be exposed with their names, and not require expressions to prefix them with  `properties`. **Everything else should be fully-qualified? How to do more deeply-nested queries and those on lists is an open question. How to disambiguate names that could appear in multiple places in an Item?** 
+
+(TBD: there is a proposal to allow finding what queryables are available for a subset of collections, e.g., `/queryables?collections=c1,c3`).  
+
+The Landing Page endpoint (`/`) will have a Link with rel `http://www.opengis.net/def/rel/ogc/1.0/queryables` with an href to the endpoint `/queryables` (TBD: hopefully, this will have a parameter `collections` defined on it to query for the intersection of the queryables in a group of collections). Additionally, each Collection resource will have a Link to the queryables resource for that collection, e.g., `/collections/collection1/queryables`. 
+
+The queryables endpoint returns a JSON Schema describing the names and types variables that may be used in filter expressions. This response is defined in JSON Schema because it is a well-specified typed schema, but it is not used for validating a JSON document derived from it. This schema defines the variables that may be used in a CQL filter.
```

Comment:
Might be nice to explain the 'simple' path to doing this. Like the next sections explain all these complex possibilities, that make it seem a bit scary to implement this whole extension. I think it'd be good to be clear about the core, minimum that a STAC implementor needs to do in order to get Filter extension to work. Do they need to make every field queryable? Is there a minimum implementation that they could almost hard code that returns datetime, geometry, other required fields? Could they hardcode the common metadata that 'might' appear? Or easily code up something that checks common metadata and includes it, but isn't trying to construct JSON Schemas on the fly from arbitrary STAC data?

I want to think about how we make an easy onramp for a STAC implementor, so they can do the minimum and have it be valuable. Perhaps put some stuff in an 'advanced' section.

Reply:
I'll do that.